### ISSUE TITLE
animationFrame() has memory leak.

### ISSUE THREAD

**Issue by C-zhui** [ReactiveX/rxjs#6946]
### Describe the bug

animationFrame will add subscriptions again again before unsubscribe.

### Expected behavior

Here I describe the problem.
I can't find the logic of removing sub subscription inside `animationFrame`.
So I add this to source code in `Subscription` class.
![image](https://user-images.githubusercontent.com/24366562/164989061-0198045b-83e7-483b-a984-903788440ed1.png)
Add this to project code.
![image](https://user-images.githubusercontent.com/24366562/164989078-3c4790cb-549b-4935-b78f-adf78f94d447.png)
You will see this.
![image](https://user-images.githubusercontent.com/24366562/164988919-e40a1b17-5269-4a77-8775-f17464bcc640.png)
shouldn't they be remove from `array:this._finalizers` over time???

### Reproduction code

```typescript
As shown in the prev section.
```

### Version

7.5.5

**Comment by ajafff** [comment]
I can confirm that it accumulates a new Subscription for each animation frame. The Subscriptions returned from `animationFrameProvider.schedule` are never closed/unsubscribed, therefore they are not removed from the list of finalizers.

I tried fixing this, but I don't know which is the preferred solution:
* don't collect all subscriptions in a parent subscription in `animationFramesFactory`, instead save the latest subscription in a local variable and return a finalizer function that unsubscribes from that subscription
* in `animationFrameProvider.schedule`, unsubscribe from the returned subscription after calling the callback
* don't use `animationFrameProvider.schedule` at all, because creating a Subscription for every invocation seems unnecessary (wasteful and complex). instead just use `animationFrameProvider.requestAnimationFrame`

**Comment by C-zhui** [comment]
Yeah, generating sub subscriptions and replacing in turn is unnecessary, I may vote solution 3 ( directly use raf ).
It's crazy that there is such a hidden array growing over time. And suddenly stop the world to resize itself to store more items for nothing. It should be fix as soon as possible.

**Comment by DavidWeiss2** [comment]
In my opinion:
> * in `animationFrameProvider.schedule`, unsubscribe from the returned subscription after calling the callback

**Comment by icoco** [comment]
have to hard code to free ?  question : https://github.com/ReactiveX/rxjs/discussions/7523
